### PR TITLE
feat: add explain rest api for sql_rule

### PR DESCRIPTION
**Pull request by xuyukeviki**
See https://or28elw3b5.feishu.cn/docx/WZoedzBceoqgNBxoEbUcevrTntf for details

### CODE REVIEW

**Comment by ngjaying** [comment]
Thanks @xuyukeviki. Please follow the DCO check link to pass it.
Additionally, we prefer linear history, so please rebase your commits. Check https://github.com/lf-edge/ekuiper/blob/master/docs/en_US/CONTRIBUTING.md for detail.

**Comment by xuyukeviki** [comment]
> Thanks @xuyukeviki. Please follow the DCO check link to pass it. Additionally, we prefer linear history, so please rebase your commits. Check https://github.com/lf-edge/ekuiper/blob/master/docs/en_US/CONTRIBUTING.md for detail.

done

**Comment by ngjaying** [comment]
> > Thanks @xuyukeviki. Please follow the DCO check link to pass it. Additionally, we prefer linear history, so please rebase your commits. Check https://github.com/lf-edge/ekuiper/blob/master/docs/en_US/CONTRIBUTING.md for detail.
> 
> done

DCO is still not passed. Try not to use merge commit, use rebase instead.

File: internal/topo/planner/logicalPlan.go [hunk 1]
```diff
@@ -27,10 +35,58 @@ type LogicalPlan interface {
...
+}
+
+func (p *baseLogicalPlan) BuildExplainInfo(id int64) {
+	p.self.BuildExplainInfo(id)
+}
+
+func (p *baseLogicalPlan) Type() string {
+	ty := reflect.TypeOf(p.self)
+	s := strings.Split(ty.String(), ".")
+	return s[1]
```
Comment (Yisaer) [inline internal/topo/planner/logicalPlan.go:77]: I don't think `reflect` is a proper way to get the type string, I tend to initialize the type string during init logical plan like tidb

https://github.com/pingcap/tidb/blob/6888a1aa6b0f31b1c23e28bc7811bcdf2c5e5ddf/planner/core/initialize.go#L41-L44

**Review by Yisaer — CHANGES_REQUESTED** [review]
Could u split the pr into serval pr and write the unit test for the explain info of logical plans?

**Review by Yisaer — CHANGES_REQUESTED** [review]
Could u split the pr into serval pr and write the unit test for the explain info of logical plans?

**Pull request closed without merging**